src/Lti/Collection/LtiInstanceCollection.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Lti\Collection;
+
+use ArrayIterator;
+use Countable;
+use IteratorAggregate;
+use OAT\SimpleRoster\Entity\LtiInstance;
+use OutOfBoundsException;
+use Traversable;
+
+class LtiInstanceCollection implements Countable, IteratorAggregate
```

Comment:
:+1: 
But makes no sense, if you look at LtiInstance XML entity definition -> label and link are unique properties, so I don't see a way, to get real duplicates from DB

Reply:
What we have in database is not relevant here, you could still add the same object multiple times. Regarding the naming, I think it's not necessary to note that it's a unique collection, only if we would use non unique collections as well (but it's not the case).

Reply:
My only concern here is that it can cause misunderstanding. Just Imagine you creating this class with 10 instances and then, for some reason, when you call the count method it returns only 2. It seems a bit strange for me because it's happening something more that it's not informed to the client. :(

Reply:
Allright, I renamed the class 👍